tensor2tensor/rl/envs/simulated_batch_env.py

```diff
@@ -112,20 +114,22 @@ def __init__(self, hparams, length,
     _, self.action_shape, self.action_dtype = get_action_space(environment_spec)
 
     # TODO(lukaszkaiser): do this in a more cleaner way
+    # remove other_hparams
+    hparams = copy.copy(other_hparams)
     hparams.video_num_input_frames, hparams.video_num_target_frames = (
-        hparams.model_hparams.video_num_input_frames,
-        hparams.model_hparams.video_num_target_frames)
+        hparams.environment_spec.video_num_input_frames,
```

Comment:
Why not take it from model hparams?

Reply:
The idea is that environment_spec should have all data needed to instantiate SimulateBatchEnv. Otherwise this are getting really confusing. It is possible that my choice is not the best, for example may be we should have environment_spec.model field?

Reply:
My intention is that environment_spec has all information needed to instantiate SimulatedBatchEnv. Otherwise, I found it hard to understand what the code really does.

Reply:
But it breaks the stochastic test. At this point this code is impossible to merge as it breaks tests and you're blocking Mohammad since the whole loop stopped working after the refactor. So we need to try to merge this ASAP and check if it works, or just revert the whole refactor to go back to a working state.